server/src/main/java/io/littlehorse/common/model/metadatacommand/subcommand/ScheduleWfRunCommandModel.java

```diff
@@ -0,0 +1,164 @@
+            Integer majorVersion,
+            Integer revision,
+            WfRunIdModel parentWfRunId,
+            Map<String, VariableValueModel> variables,
+            String cronExpression) {
+        this.scheduledId = scheduledId;
+        this.wfSpecName = wfSpecName;
+        this.majorVersion = majorVersion;
+        this.revision = revision;
+        this.variables = variables;
+        this.cronExpression = cronExpression;
+        this.parentWfRunId = parentWfRunId;
+    }
+
+    @Override
+    public void initFrom(Message proto, ExecutionContext context) throws LHSerdeError {
+        ScheduleWfRun p = (ScheduleWfRun) proto;
+        scheduledId = LHSerializable.fromProto(p.getScheduledId(), ScheduledWfRunIdModel.class, context);
+        wfSpecName = p.getWfSpecName();
+        if (p.hasMajorVersion()) {
+            majorVersion = p.getMajorVersion();
+        }
+        if (p.hasRevision()) {
+            revision = p.getRevision();
+        }
+        if (p.hasParentWfRunId()) {
+            parentWfRunId = LHSerializable.fromProto(p.getParentWfRunId(), WfRunIdModel.class, context);
+        }
+        for (Map.Entry<String, VariableValue> e : p.getVariablesMap().entrySet()) {
+            variables.put(e.getKey(), VariableValueModel.fromProto(e.getValue(), context));
+        }
+        cronExpression = p.getCronExpression();
+    }
+
+    @Override
+    public ScheduleWfRun.Builder toProto() {
+        ScheduleWfRun.Builder out = ScheduleWfRun.newBuilder()
+                .setScheduledId(scheduledId.toProto())
+                .setWfSpecName(wfSpecName)
+                .setCronExpression(cronExpression);
+        if (majorVersion != null) {
+            out.setMajorVersion(majorVersion);
+        }
+        if (revision != null) {
+            out.setRevision(revision);
+        }
+        if (parentWfRunId != null) {
+            out.setParentWfRunId(parentWfRunId.toProto());
+        }
+
+        for (Map.Entry<String, VariableValueModel> e : variables.entrySet()) {
+            out.putVariables(e.getKey(), e.getValue().toProto().build());
+        }
+        return out;
+    }
+
+    @Override
+    public Class<ScheduleWfRun> getProtoBaseClass() {
+        return ScheduleWfRun.class;
+    }
+
+    @Override
+    public boolean hasResponse() {
+        return false;
+    }
+
+    @Override
+    public Message process(ProcessorExecutionContext executionContext, LHServerConfig config) {
+        ScheduledWfRunModel scheduledWfRun = executionContext.getableManager().get(scheduledId);
+        if (scheduledWfRun != null) {
+            CommandModel commandToExecute = new CommandModel(createRunWfCommand(executionContext), new Date());
+            LHTimer runWfTimer = new LHTimer(commandToExecute);
+            runWfTimer.maturationTime = new Date();
+            executionContext.getTaskManager().scheduleTimer(runWfTimer);
+            Optional<Date> scheduledTime = LHUtil.nextDate(cronExpression, new Date());
```

Comment:
I see one problem here—what if we have a cron expression to run it every second, and then we have a little bit of lag? We could "drop" some instances.

Should we keep a `last_wf_run_timestamp` field in the `ScheduledWfRun` and use that instead of `new Date()`?

Reply:
For example if someone is using a LH `ScheduledWfRun` to send a billing metric to Orb that says "hey! the customer used the service for the last minute" and we have some downtime for 10 minutes for some reason, we would want to ensure that 10 workflows run when the system comes back online rather than only running 1.